packages/@react-spectrum/s2/src/ComboBox.tsx

```diff
@@ -383,11 +431,333 @@ export function ComboBoxSection<T extends object>(props: ComboBoxSectionProps<T>
   return (
     <>
       <AriaListBoxSection
-        {...props}
-        className={section({size})}>
+        {...props}>
         {props.children}
       </AriaListBoxSection>
-      <Divider />
+      <Divider size={size} />
+    </>
+  );
+}
+
+// TODO: not quite sure why typescript is complaining when I types this as T extends object
+const ComboboxInner = forwardRef(function ComboboxInner(props: ComboBoxProps<any> & {isOpen: boolean}, ref: ForwardedRef<TextFieldRef | null>) {
```

Comment:
Mainly a reorg of code + adding loader logic/spinner rendering, for the most part most of this is the same as before